tests/test_item.py

```diff
@@ -33,15 +36,18 @@ class TestItem(Item):
         i4 = TestItem(i3)
         self.assertEqual(i4["name"], "john doe")
 
-        self.assertRaises(KeyError, TestItem, {"name": "john doe", "other": "foo"})
+        with pytest.raises(KeyError):
+            TestItem({"name": "john doe", "other": "foo"})
 
     def test_invalid_field(self):
         class TestItem(Item):
             pass
 
         i = TestItem()
-        self.assertRaises(KeyError, i.__setitem__, "field", "text")
-        self.assertRaises(KeyError, i.__getitem__, "field")
+        with pytest.raises(KeyError):
+            i.__setitem__("field", "text")
```

Comment:
```suggestion
            i["field"] = "text"
```

I think this style corresponds better to how users write the code, i.e. to what we really want to test.